examples/runc-hook/deploy.yaml

```diff
@@ -0,0 +1,140 @@
+kind: ServiceAccount
+metadata:
+  name: runc-hook
+  namespace: runc-hook
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: runc-hook-cluster-role
+rules:
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "watch", "list"]
+- apiGroups: ["security.openshift.io"]
+  # It is necessary to use the 'privileged' security context constraints to be
+  # able mount host directories as volumes, use the host networking, among others.
+  # This will be used only when running on OpenShift:
+  # https://docs.openshift.com/container-platform/4.9/authentication/managing-security-context-constraints.html#default-sccs_configuring-internal-oauth
+  resources: ["securitycontextconstraints"]
+  resourceNames: ["privileged"]
+  verbs: ["use"]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: runc-hook-cluster-role-binding
+subjects:
+- kind: ServiceAccount
+  name: runc-hook
+  namespace: runc-hook
+roleRef:
+  kind: ClusterRole
+  name: runc-hook-cluster-role
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: hook-config
+  namespace: runc-hook
+data:
+  prestart.sh: |
+    #!/bin/sh
+    (cat ; echo) >> /tmp/hooks.log
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: runc-hook
+  namespace: runc-hook
+  labels:
+    k8s-app: runc-hook
+spec:
+  selector:
+    matchLabels:
+      k8s-app: runc-hook
+  template:
+    metadata:
+      labels:
+        k8s-app: runc-hook
+      annotations:
+        container.apparmor.security.beta.kubernetes.io/gadget: "unconfined"
+    spec:
+      serviceAccount: runc-hook
+      hostPID: true
+      hostNetwork: true
+      containers:
+      - name: gadget
+        terminationMessagePolicy: FallbackToLogsOnError
+        image: gadget-runc-hook:latest
+        imagePullPolicy: Always
+        env:
+          - name: NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
+          - name: POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+        securityContext:
+          capabilities:
+            add:
+              # fanotify requires CAP_SYS_ADMIN
+              - SYS_ADMIN
+
+              # Needed by setrlimit
+              - SYS_RESOURCE
+
+              # Needed for gadgets that don't dumb the memory rlimit.
+              - IPC_LOCK
+        volumeMounts:
+        - name: host
+          mountPath: /host
+        - name: run
+          mountPath: /run
+        - name: modules
+          mountPath: /lib/modules
+        - name: debugfs
+          mountPath: /sys/kernel/debug
+        - name: cgroup
+          mountPath: /sys/fs/cgroup
+        - name: bpffs
+          mountPath: /sys/fs/bpf
+        - name: hook-config
+          mountPath: /hooks
+      tolerations:
+      - effect: NoSchedule
+        operator: Exists
+      - effect: NoExecute
+        operator: Exists
```

Comment:
Do we really want this? Why?

Reply:
Inspektor Gadget has this so we can inspect the master nodes too. Should I remove it?